hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    __rest_api_response = utils.HSRestAPIResponse()
+
+    @property
+    def allowed_methods(self):
+        return ['GET', 'POST', 'DELETE']
+
+    def get(self, request, pk, filename):
+        # check resource exists for the provided resource id (pk)
+        #utils.get_resource(resource_id=pk)
+
+        # check the user is authorized for this action and if the resource exists for the given resource id (pk)
+        utils.authorize(request, pk, view=True)
+        try:
+            f = hydroshare.get_resource_file(pk, filename)
+        except:
+            err_msg = 'File with file name {file_name} does not exist for resource with resource id ' \
+                      '{res_id}'.format(file_name=filename, res_id=pk)
+            raise NotFound(detail=err_msg)
+
+        # redirects to django_irods/views.download function
+        return HttpResponseRedirect(f.url)
+
+    def post(self, request, pk):
+        #resource = utils.get_resource(resource_id=pk)
+        # check the user is authorized for this action and if the resource exists for the given resource id (pk)
+        resource, _, _ = utils.authorize(request, pk, edit=True)
+        resource_files = request.FILES.values()
+        if len(resource_files) == 0:
+            error_msg = {'missing_file': 'No file was provided to add to the resource.'}
+            raise ValidationError(detail=error_msg)
+
+        # TODO: user utils.validate_resource_file_size() here
+        valid = hydroshare.check_resource_files(resource_files)
+        if not valid:
+            error_msg = {
+                'file_size_error': 'The resource file is larger than the supported size limit %s. '
+                                    % file_size_limit_for_display
+            }
+            raise ValidationError(detail=error_msg)
+
+        extract_metadata = request.data.get('extract_metadata', 'false')
+        if extract_metadata.lower() not in ('true', 'false'):
+            error_msg = {'extract_metadata': 'Must have a boolean value of true or false.'}
+            raise ValidationError(detail=error_msg)
+
+        if extract_metadata.lower() == 'true':
+            extract_metadata = True
+        else:
+            extract_metadata = False
+
+        error_msg = utils.resource_file_add_preprocess(resource=resource, files=[resource_files[0]], user=request.user,
+                                                       extract_metadata=extract_metadata)
+        if error_msg:
+            raise ValidationError(detail=error_msg)
+        try:
+            error_msg, res_file_objects = utils.resource_file_add_postprocess(resource=resource, files=[resource_files[0]],
+                                                                   user=request.user, extract_metadata=extract_metadata)
+        except Exception as ex:
+            raise ValidationError(detail=ex.message)
```

Comment:
Yes, it is validating the first file. Should we limit one file upload at a time and throw error in case of multiple file upload? I think that would make file validation and upload easier.